### PR TITLE
Chris - Chitter

### PR DESCRIPTION
**Pull request by Cwisfa**
# Your name

Chris Andrews

# User stories 

Please list which user stories you've implemented (delete the ones that don't apply).

- [X] User story 1: "I want to see all the messages (peeps) in a browser"
- [X] User story 2: "I want to post a message (peep) to chitter"
- [ ] User story 3: "I want to see the date the message was posted"
- [ ] User story 4: "I want to see a list of peeps in reverse chronological order"
- [ ] User story 5: "I want to filter on a specific keyword"

# README checklist

Does your README contains instructions for

- [ ] how to install,
- [ ] how to run,
- [ ] and how to test your code?

[Here is a pill](https://github.com/makersacademy/course/blob/main/pills/readmes.md) that can help you write a great README!

### CODE REVIEW

File: lib/peeps.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require 'pg'
+
+class Peeps
+  def self.all
+    if ENV["ENVIRONMENT"] == "test"
+      connection = PG.connect(dbname: "chitter_test")
```
Comment (lucyydotp) [inline lib/peeps.rb:6]: the connection should be made when the class is instantiated and kept as a reference instead of creating a new one for each query

Comment (lucyydotp) [reply lib/peeps.rb:6]: probably should also take the env stuff out and pass it in through the constructor

File: lib/peeps.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def self.create(message)
+    if ENV['ENVIRONMENT'] == 'test'
+      connection = PG.connect(dbname: 'chitter_test')
+    else
+      connection = PG.connect(dbname: 'chitter')
+    end
+
+    connection.exec("INSERT INTO peeps (message) VALUES('#{message}')")
```
Comment (lucyydotp) [inline lib/peeps.rb:22]: use a prepared statement to avoid sql injection

File: spec/features/create_peep_spec.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+feature "Adds a new peep" do
+    scenario "User makes peep, posts it to Peeps" do
+        connection = PG.connect(dbname: "chitter_test")
```
Comment (lucyydotp) [inline spec/features/create_peep_spec.rb:3]: this doesn't get used

File: views/new_peep.erb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    border-color: black;
+    border-width: 10px;
+}
+
+h3 {
+ font-family: "Gill Sans", sans-serif;
+
+}
+
+}
```
Comment (lucyydotp) [inline views/new_peep.erb:28]: I can't see an opening brace for this

File: views/new_peep.erb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+<h1>- CHITTER -</h1>
+<form action="/peeps" method="post">
+  <input type="text" name="peep" />
+  <input type="submit" value="Send peep!" />
+</form> 
+
+<style>
+body {
+background-color: lightblue;
```
Comment (lucyydotp) [inline views/new_peep.erb:11]: try and keep your indentation consistent, it'll help keep your code readable

File: views/peeps.erb [hunk 6]
```diff
@@ -0,0 +1,36 @@
+<h1>- CHITTER -</h1>
+<h3>Latest Peeps </h3>
+<ul>
+  <% @peeps.each do |peeps| %>
```
Comment (lucyydotp) [inline views/peeps.erb:4]: `peeps` should be called `peep`, there's only one

File: app.rb [hunk 7]
```diff
@@ -1,9 +1,24 @@
-require 'sinatra/base'
+require "sinatra/base"
+require "./lib/peeps.rb"
 
 class Chitter < Sinatra::Base
-  get '/test' do
-    'Test page'
+  get "/test" do
+    "Test page"
   end
```
Comment (oceansize) [inline app.rb:7]: I probably would have deleted this route as it's just here to get you started ☺️

File: app.rb [hunk 8]
```diff
@@ -1,9 +1,24 @@
...
 
+  get "/peeps" do
```
Comment (EdwardAndress) [inline app.rb:9]: Nice restful routing

**Comment by EdwardAndress** [comment]
Hey Chris, you've made a good start here and the work. you've done demonstrates the main learning objectives around reading from and writing to a database. If you had a little more time, would you have been able to get started on the next user story?

**Pull request closed without merging**